### PR TITLE
fix: non deterministic cost

### PR DESCRIPTION
**Pull request by iammadab**
## Issue being fixed or feature implemented

Cost values depended on the ordering of key values in the rocksdb storage layer, leading to different state roots when prefix computation changes. 

## What was done?

Unified the costs, regardless of the ordering in the underlying storage layer. 
- if the next key is from another merk, rather than add the cost of the key we add the max key length
- if the next key is none, add the max key length rather than nothing. 

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by fominok — APPROVED** [review]
gentlemen, we did it

File: storage/src/rocksdb_storage/storage_context/raw_iterator.rs [hunk 1]
```diff
@@ -102,16 +105,23 @@ impl<'a> RawIterator for PrefixedRocksDbRawIterator<DBRawIteratorWithThreadMode<
...
-            } else {
+        let value = match self.raw_iterator.key() {
+            Some(k) => {
+                // Even if we truncate prefix, loaded cost should be maximum for the whole
+                // function
+                if k.starts_with(&self.prefix) {
+                    cost.storage_loaded_bytes += k.len() as u32;
+                    Some(k.split_at(self.prefix.len()).1)
+                } else {
+                    cost.storage_loaded_bytes += MAX_PREFIXED_KEY_LENGTH;
```
Comment (QuantumExplorer) [inline storage/src/rocksdb_storage/storage_context/raw_iterator.rs:116]: Add a comment explaining this situation.

File: storage/src/rocksdb_storage/storage_context/raw_iterator.rs [hunk 2]
```diff
@@ -102,16 +105,23 @@ impl<'a> RawIterator for PrefixedRocksDbRawIterator<DBRawIteratorWithThreadMode<
...
+                    None
+                }
+            }
+            None => {
+                cost.storage_loaded_bytes += MAX_PREFIXED_KEY_LENGTH;
```
Comment (QuantumExplorer) [inline storage/src/rocksdb_storage/storage_context/raw_iterator.rs:121]: Add a comment explaining this situation.

File: storage/src/rocksdb_storage/storage_context/raw_iterator.rs [hunk 3]
```diff
@@ -122,10 +132,18 @@ impl<'a> RawIterator for PrefixedRocksDbRawIterator<DBRawIteratorWithThreadMode<
         self.raw_iterator
             .key()
             .map(|k| {
-                cost.storage_loaded_bytes += k.len() as u32;
-                k.starts_with(&self.prefix)
+                if k.starts_with(&self.prefix) {
+                    cost.storage_loaded_bytes += k.len() as u32;
+                    true
+                } else {
+                    cost.storage_loaded_bytes += MAX_PREFIXED_KEY_LENGTH;
```
Comment (QuantumExplorer) [inline storage/src/rocksdb_storage/storage_context/raw_iterator.rs:139]: Add a comment explaining this situation.

File: storage/src/rocksdb_storage/storage_context/raw_iterator.rs [hunk 4]
```diff
@@ -122,10 +132,18 @@ impl<'a> RawIterator for PrefixedRocksDbRawIterator<DBRawIteratorWithThreadMode<
...
+                    false
+                }
+            })
+            .unwrap_or_else(|| {
+                cost.storage_loaded_bytes += MAX_PREFIXED_KEY_LENGTH;
```
Comment (QuantumExplorer) [inline storage/src/rocksdb_storage/storage_context/raw_iterator.rs:144]: Add a comment explaining this situation.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]
We need more comments.

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**